### PR TITLE
Enhance xds logging

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

To  facilitate xds connection error debugging

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @shonecyx @ramaraochavali

File: pkg/istio-agent/xds_proxy.go [hunk 1]
```diff
@@ -693,6 +693,13 @@ func sendUpstream(upstream xds.DiscoveryClient, request *discovery.DiscoveryRequ
 
 // sendDownstream sends discovery response.
 func sendDownstream(downstream adsStream, response *discovery.DiscoveryResponse) error {
+	tStart := time.Now()
+	defer func() {
+		// This is a hint to help debug slow responses.
+		if time.Since(tStart) > 10*time.Second {
```
Comment (keithmattix) [inline pkg/istio-agent/xds_proxy.go:699]: This feels generally useful for responses that are less than 10s; maybe set this to a flag of some sort?

**Review by keithmattix — APPROVED** [review]
Seems reasonable to me

File: pkg/istio-agent/xds_proxy.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/istio-agent/xds_proxy.go:699]: I feel same when i add this log

Comment (ramaraochavali) [reply pkg/istio-agent/xds_proxy.go:699]: What impacts we have with slower responses? Isn't send time on istiod measure this?

Comment (hzxuzhonghu) [reply pkg/istio-agent/xds_proxy.go:699]: No impact, just a warning now.

**Pull request merged**